R/sitrep.R

```diff
@@ -2,13 +2,17 @@
 #' @importFrom memoise memoise
 NULL
 
-check_for_rstudio_updates <- function(os = tolower(Sys.info()[["sysname"]]), version = rstudioapi::getVersion(), in_rstudio = rstudioapi::isAvailable()) {
-
+check_for_rstudio_updates <- function(os = tolower(Sys.info()[["sysname"]]),
+                                      version = rstudioapi::versionInfo()$long_version,
+                                      in_rstudio = rstudioapi::isAvailable()) {
   if (!in_rstudio) {
     return()
   }
 
-  url <- sprintf("https://www.rstudio.org/links/check_for_update?version=%s&os=%s&format=%s", version, os, "kvp")
+  url <- sprintf(
+    "https://www.rstudio.org/links/check_for_update?version=%s&os=%s&format=%s",
+    utils::URLencode(version, reserved = TRUE), os, "kvp"
+  )
```

Comment:
What's new in the URL:

``` r
version <- rstudioapi::versionInfo()$long_version
version
#> [1] "2021.09.1+372"
URLencode(version, reserved = TRUE)
#> [1] "2021.09.1%2B372"
```

Reply:
Two issues I see, unfortunately

1) Older versions of RStudio didn't have a `long_version`, only a `version`. That would be okay, as you could potentially fallback to `version` if `long_version` is not found, but
2) calling URLencode on a version number from `version` rather than `long_version` gives me an error. For example

```
rstudioapi::versionInfo()
$citation

To cite RStudio in publications use:

  RStudio Team (2021). RStudio: Integrated Development Environment for R. RStudio, PBC, Boston, MA URL http://www.rstudio.com/.

A BibTeX entry for LaTeX users is

  @Manual{,
    title = {RStudio: Integrated Development Environment for R},
    author = {{RStudio Team}},
    organization = {RStudio, PBC},
    address = {Boston, MA},
    year = {2021},
    url = {http://www.rstudio.com/},
  }


$mode
[1] "desktop"

$version
[1] ‘1.4.1717’

$release_name
[1] "Juliet Rose"

> URLencode(rstudioapi::versionInfo()$version, reserved=TRUE)
Error in strsplit(URL, "") : non-character argument
```

Reply:
Converting $version to as.character first should work though

Reply:
I tested your function out in RStudio 1.4.1717, and it doesn't return the expected results, but does if I add this to the URL: `&manual=true` after `format=kvp`-- when you click on `Help > Check for Updates` it calls the url with the 
`version={version}&os={os}&format=kvp&manual=true` format, so that may also be required